examples/Clients/Mailer/Program.cs

```diff
@@ -0,0 +1,83 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Mail;
+using Common;
+using Grpc.Core;
+
+namespace Sample.Clients
+{
+    class Program
+    {
+        static async Task<int> Main(string[] args)
+        {
+            // Server will only support Https on Windows and Linux
+            var credentials = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? ChannelCredentials.Insecure : ClientResources.SslCredentials;
+
+            var channel = new Channel("localhost:50051", credentials);
+
+            Console.WriteLine("Connecting to mailbox...");
+            Console.WriteLine();
+
+            await channel.ConnectAsync();
+
+            Console.WriteLine("Connected");
+            Console.WriteLine("Press escape to exit. Press any other key to forward mail.");
+
+            var client = new Mailer.MailerClient(channel);
+            using (var mailbox = client.Mailbox(headers: new Metadata { new Metadata.Entry("mailbox-name", "DefaultMailBox") }))
```

Comment:
So this would never be configurable anyway? Maybe make this an optional command line arg?